package topicmd;

message Field 
{
  optional string field_name = 1 [default = "@body"];
  repeated int32 token_id = 2;
  repeated int32 token_count = 3;
}

message Item
{
  optional int32 id = 1;
  repeated Field field = 2;
}

message Batch {
  repeated string token = 1;
  repeated Item item = 3;
}

message DataLoaderConfig {
  required int32 instance_id = 1;
  optional string disk_path = 2;  
  optional int32 queue_size = 3 [default = 10];
}

message InstanceConfig {
  optional int32 processors_count = 1 [default = 1];
}

message ModelConfig {
  optional int32 topics_count = 2 [default = 32];
  optional bool enabled = 3 [default = false];
  optional int32 inner_iterations_count = 4 [default = 10];
  optional string field_name = 5 [default = "@body"];
}

message LoggerConfig {
  optional string log_location = 1;

  enum Level {
    DISABLED = 0;
    ERROR = 1;
    WARNING = 2;
    INFO = 3;
    VERBOSE = 4;
  }

  optional Level level = 2 [default = INFO];
}

message ItemTopics {
  optional int32 id = 1;
  repeated float topic_weight = 2;
}

message BatchTopics {
  repeated ItemTopics item_topics = 1;
}

message TokenTopics {
  optional string token = 1;
  optional int32 token_id = 2;
  repeated float topic_weight = 3;
}

message ModelTopics {
  repeated TokenTopics token_topic = 1;
}
